open-craft/ansible-mysql
# Raise limit of open files

Testing:

1. Download https://github.com/open-craft/deployment-deploy-databases
2. Create virtualenv, install from requirements.txt, activate it
3. Follow instructions from that repo (until: ansible-galaxy)
4. Replace the internal ansible-mysql with the one of this branch
5. Get an OpenStack VM or any other, with systemd (e.g. 16.04). Different versions will produce different results because the mysql roles might have small incompatibilites
6. Adapt parameters and run:
   ```
    ~/.virtualenvs/myvirtualenv/bin/ansible-playbook deploy-all.yml -i "149.202.190.81," -e MYSQL_OPENSTACK_DB_DEVICE=/dev/sdb1 -e MYSQL_OPEN_FILES_LIMIT=1420 -e MYSQL_PROCESSES_LIMIT=857 -u ubuntu # --extra-vars @private-extra-vars.yml -vv
   ```
7. Deployment should not fail because of the limits. Actually, expect it to fail for other things: python version, partitions, mysql root password, … Some of the roles we're using might need to be updated to work smoothly
8. Log in through SSH
9. ~`ulimit -n`, should show the number you chose for number of files, und `ulimit -u` the one for number of processes~ You don't need to check ulimit because the role now doesn't change the numbers for interactive processes like `bash`
10. Find mysqld's PID, and `cat /proc/PID/limits`, it should show your numbers
11. Log in to mysql, and `show global variables like 'open%';`, it should show your number
